front/src/modules/rollingStock/components/RollingStockEditor/PowerRestrictionGridModal.tsx

```diff
@@ -0,0 +1,131 @@
+
+type PowerRestrictionGridModalProps = {
+  powerRestrictionsList: string[];
+  updatePowerRestrictions: (data: string) => void;
+};
+
+const PowerRestrictionGridModal = ({
+  powerRestrictionsList,
+  updatePowerRestrictions,
+}: PowerRestrictionGridModalProps) => {
+  const { closeModal } = useModal();
+  const { t } = useTranslation('rollingstock');
+
+  const [filter, setFilter] = useState('');
+  const modalRef = useRef<HTMLDivElement>(null);
+
+  // Split the list in two arrays one with code starting with letter and one with digit
+  const [digitalPowerRestrictions, letterPowerRestrictions] = useMemo(() => {
+    if (!powerRestrictionsList) return [];
+    const sortedArray = [...powerRestrictionsList].sort();
+    const splittingIndex = sortedArray.findIndex((powerRestriction) =>
+      Number.isNaN(Number(powerRestriction[0]))
+    );
+    const digitalArray = sortedArray.slice(0, splittingIndex);
+    const letterArray = sortedArray.slice(splittingIndex);
+    return [digitalArray, letterArray];
+  }, [powerRestrictionsList]);
+
+  const isNewValue = !powerRestrictionsList?.some((code) => code.includes(filter));
+
+  // Allow the user to escape the modal by pressing escape and to trap the focus inside it
+  useEffect(() => {
+    const modalElement = modalRef.current;
+
+    const focusableElements = modalElement?.querySelectorAll(
+      // last declaration stands for all elements not natively focusable like li
+      'input, button, [tabindex]:not([tabindex="-1"])'
+    ) as NodeListOf<Element>;
+
+    const firstElement = focusableElements[0] as HTMLElement;
+    const lastElement = focusableElements[focusableElements?.length - 1] as HTMLElement;
+
+    /**
+     *
+     * Prevent the tab event and set focus on :
+     * - last element if we are pressing on "shift" in addition to "tab" and are on the first element
+     * - first element if we are only pressing "tab" and are on the last element
+     */
+    const handleTabKeyPress = (event: KeyboardEvent) => {
+      if (event.key === 'Tab') {
+        if (event.shiftKey && document.activeElement === firstElement) {
+          event.preventDefault();
+          lastElement.focus();
+        } else if (!event.shiftKey && document.activeElement === lastElement) {
+          event.preventDefault();
+          firstElement.focus();
+        }
+      }
+    };
+
+    const handleEscapeKeyPress = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        closeModal();
+      }
+    };
+
+    modalElement?.addEventListener('keydown', handleTabKeyPress);
+    modalElement?.addEventListener('keydown', handleEscapeKeyPress);
+
+    return () => {
+      modalElement?.removeEventListener('keydown', handleTabKeyPress);
+      modalElement?.removeEventListener('keydown', handleEscapeKeyPress);
+    };
+  }, []);
+
+  return (
+    <div className="power-restrictions-modal p-2" ref={modalRef}>
```

Comment:
I think this class is not used
```suggestion
    <div className="p-2" ref={modalRef}>
```

Reply:
done, good catch !